mava/wrappers/jumanji.py

```diff
@@ -117,3 +124,96 @@ def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
 
         # Aggregate the list of individual rewards and use a single team_reward.
         return self.aggregate_rewards(timestep, modified_observation)
+
+
+class ConnectorWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the MA Connector environment.
+
+    Do not use the AgentID wrapper with this env, it has implicit agent IDs.
+    """
+
+    def __init__(self, env: MaConnector, has_global_state: bool = False):
+        super().__init__(env)
+        self.has_global_state = has_global_state
+
+    def modify_timestep(
+        self, timestep: TimeStep
+    ) -> TimeStep[Union[Observation, ObservationGlobalState]]:
+        """Modify the timestep for the Connector environment."""
+
+        # TARGET = 3 = The number of different types of items on the grid.
+        def create_agents_view(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % TARGET == POSITION, True, False)
```

Comment:
Bit of a naive question here, but are we allowing all agents to always see where all goals for all other agents are? Just to double check that is is intended if it is the case @sash-a?

Reply:
Yup, this is how connector works :+1:

Reply:
Spoke offline. This makes sense to me for now.